core/src/main/java/org/apache/carbondata/core/util/path/TrashUtil.java

```diff
@@ -0,0 +1,201 @@
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.carbondata.common.logging.LogServiceFactory;
+import org.apache.carbondata.core.constants.CarbonCommonConstants;
+import org.apache.carbondata.core.datastore.filesystem.CarbonFile;
+import org.apache.carbondata.core.datastore.impl.FileFactory;
+import org.apache.carbondata.core.exception.CarbonFileException;
+import org.apache.carbondata.core.util.CarbonUtil;
+
+import org.apache.commons.io.FileUtils;
+
+import org.apache.hadoop.fs.permission.FsAction;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.io.IOUtils;
+import org.apache.log4j.Logger;
+
+/**
+ * Mantains the trash folder in carbondata. This class has methods to copy data to the trash and
+ * remove data from the trash.
+ */
+public final class TrashUtil {
+
+  private static final Logger LOGGER =
+          LogServiceFactory.getLogService(CarbonUtil.class.getName());
+
+  /**
+   * The below method copies the complete a file to the trash folder. Provide necessary
+   * timestamp and the segment number in the suffixToAdd  variable, so that the proper folder is
+   * created in the trash folder.
+   *
+   * @param carbonTablePath table path of the carbon table
+   * @param pathOfFileToCopy the files which are to be moved to the trash folder
+   * @param suffixToAdd timestamp, partition folder(if any) and segment number
+   * @return
+   */
+  public static void copyDataToTrashFolderByFile(String carbonTablePath, String pathOfFileToCopy,
+      String suffixToAdd) {
+    String trashFolderPath = CarbonTablePath.getTrashFolderPath(carbonTablePath) +
+        CarbonCommonConstants.FILE_SEPARATOR + suffixToAdd;
+    CarbonFile carbonFile  = FileFactory.getCarbonFile(pathOfFileToCopy);
+    DataOutputStream dataOutputStream = null;
+    DataInputStream dataInputStream = null;
+    try {
+      if (carbonFile.exists()) {
+        // FileUtils.copyFileToDirectory(new File(pathOfFileToCopy), new File(trashFolderPath));
+        if (!new File(trashFolderPath).exists()) {
+          FileFactory.createDirectoryAndSetPermission(trashFolderPath,
+              new FsPermission(FsAction.ALL, FsAction.ALL, FsAction.ALL));
+        }
+        dataOutputStream = FileFactory
+          .getDataOutputStream(trashFolderPath + CarbonCommonConstants.FILE_SEPARATOR +
+            carbonFile.getName());
+        dataInputStream = FileFactory
+          .getDataInputStream(pathOfFileToCopy);
+        IOUtils.copyBytes(dataInputStream, dataOutputStream, CarbonCommonConstants.BYTEBUFFER_SIZE);
+        LOGGER.info("File: " + pathOfFileToCopy + " successfully copied to the trash folder: "
+            + trashFolderPath);
+      }
+    } catch (IOException exception) {
+      LOGGER.error("Unable to copy " + pathOfFileToCopy + " to the trash folder", exception);
```

Comment:
**throw the exception to stop clean this file, if not, it will delete this file by mistake**